src/TestingUtils/Microsoft.AspNetCore.Testing/src/xunit/AspNetTestMethodRunner.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Testing
+{
+    internal class AspNetTestMethodRunner : XunitTestMethodRunner
+    {
+        private readonly object[] _constructorArguments;
+
+        public AspNetTestMethodRunner(
+            ITestMethod testMethod,
+            IReflectionTypeInfo @class,
+            IReflectionMethodInfo method,
+            IEnumerable<IXunitTestCase> testCases,
+            IMessageSink diagnosticMessageSink,
+            IMessageBus messageBus,
+            ExceptionAggregator aggregator,
+            CancellationTokenSource cancellationTokenSource,
+            object[] constructorArguments)
+            : base(testMethod, @class, method, testCases, diagnosticMessageSink, messageBus, aggregator, cancellationTokenSource, constructorArguments)
+        {
+            _constructorArguments = constructorArguments;
+        }
+
+        protected override Task<RunSummary> RunTestCaseAsync(IXunitTestCase testCase)
+        {
+            if (testCase.GetType() == typeof(XunitTestCase))
```

Comment:
This is how we take control of the test execution without needing to implement a custom discoverer.

Reply:
What type would a testCase be if the test was skipped?

Reply:
A test case that's skipped is one of two things:
1. Using `[ConditionalFact]` or similar it's a **subclass** of `XunitTestCase` and we would not hijake it
2. Using `[Fact(Skip = ...)]` it would be `XunitTestCase`, we would hijack it but it would still result in a skip